### PR TITLE
Expose shadow bias to component & feature processors. 

### PR DESCRIPTION
**Pull request by invertednormal**
Shadow bias now works more consistently with various near / far shadow planes and caster positions. Bias now also affects esm shadows which helps eliminate acne in certain situations.

### CODE REVIEW

**Review by antonmic — APPROVED** [review]

File: Gems/Atom/Feature/Common/Code/Source/Shadows/ProjectedShadowFeatureProcessor.cpp [hunk 1]
```diff
@@ -265,23 +273,19 @@ namespace AZ::Render
         view->SetCameraTransform(Matrix3x4::CreateFromTransform(desc.m_transform));
 
         ShadowData& shadowData = m_shadowData.GetElement<ShadowDataIndex>(shadowProperty.m_shadowId.GetIndex());
-        shadowData.m_bias = (nearDist / farDist) * 0.1f;
+
+        // Adjust the manually set bias to a more appropriate range for the shader.
+        shadowData.m_bias = nearDist * shadowProperty.m_bias * 0.01f;
```
Comment (VickyAtAZ) [inline Gems/Atom/Feature/Common/Code/Source/Shadows/ProjectedShadowFeatureProcessor.cpp:278]: Can you add comment on why use this equation to convert the shadowProperty.m_bias to shadowData.m_bias ?

File: Gems/AtomLyIntegration/CommonFeatures/Code/Source/CoreLights/AreaLightComponentConfig.cpp [hunk 2]
```diff
@@ -18,7 +18,7 @@ namespace AZ
             if (auto serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
             {
                 serializeContext->Class<AreaLightComponentConfig, ComponentConfig>()
-                    ->Version(6) // ATOM-15654
+                    ->Version(7) // ATOM-15654
```
Comment (VickyAtAZ) [inline Gems/AtomLyIntegration/CommonFeatures/Code/Source/CoreLights/AreaLightComponentConfig.cpp:21]: Need to update the jira number here

**Review by VickyAtAZ — APPROVED** [review]

File: Gems/Atom/Feature/Common/Code/Source/Shadows/ProjectedShadowFeatureProcessor.cpp (hunk 1 above)
Comment (invertednormal) [reply Gems/Atom/Feature/Common/Code/Source/Shadows/ProjectedShadowFeatureProcessor.cpp:279]: Expanded on the comment already there.

File: Gems/AtomLyIntegration/CommonFeatures/Code/Source/CoreLights/AreaLightComponentConfig.cpp (hunk 2 above)
Comment (invertednormal) [reply Gems/AtomLyIntegration/CommonFeatures/Code/Source/CoreLights/AreaLightComponentConfig.cpp]: Good catch, done.

**Pull request merged**